docs/contributing.rst

```diff
@@ -81,47 +81,73 @@ Thank you for your help!
 Running the compiler tests
 ==========================
 
-The ``./scripts/tests.sh`` script executes most Solidity tests automatically,
-but for quicker feedback, you might want to run specific tests.
+Prerequisites
+-------------
+
+Some tests require the `evmone <https://github.com/ethereum/evmone/releases>`_
+library, others require `libz3 <https://github.com/Z3Prover/z3>`_. The test script
+tries to discover the location of the ``evmone`` library, which can be located in
+the current directory, or the ``deps`` folder in the project top level. The required
+file is called ``libevmone.so`` on Linux systems, ``evmone.dll`` on Windows systems
+and ``libevmone.dylib`` on macOS.
+
+Running the tests
+-----------------
 
 Solidity includes different types of tests, most of them bundled into the
 `Boost C++ Test Framework <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/index.html>`_ application ``soltest``.
 Running ``build/test/soltest`` or its wrapper ``scripts/soltest.sh`` is sufficient for most changes.
 
-Some tests require the ``evmone`` library, others require ``libz3``.
+The ``./scripts/tests.sh`` script executes most Solidity tests automatically,
+including those bundled into the `Boost C++ Test Framework <https://www.boost.org/doc/libs/1_69_0/libs/test/doc/html/index.html>`_ application ``soltest`` (or its wrapper ``scripts/soltest.sh``),
+as well as command line tests and compilation tests.
 
-The test system will automatically try to discover the location of the ``evmone`` library
+The test system automatically tries try to discover the location of the ``evmone`` library
 starting from the current directory. The required file is called ``libevmone.so`` on Linux systems,
 ``evmone.dll`` on Windows systems and ``libevmone.dylib`` on MacOS. If it is not found, the relevant tests
 are skipped. To run all tests, download the library from
 `Github <https://github.com/ethereum/evmone/releases/tag/v0.3.0>`_
 and either place it in the project root path or inside the ``deps`` folder.
 
-If you do not have libz3 installed on your system, you should disable the SMT tests:
-``./scripts/soltest.sh --no-smt``.
+If the ``evmone`` library is not found, tests that use it are skipped. These tests
+are ``libsolididty/semanticTests``, ``libsolidity/GasCosts``, ``libsolidity/SolidityEndToEndTest``
+part of the soltest suite.
 
-``./build/test/soltest --help`` has extensive help on all of the options available.
-See especially:
+If the ``libz3`` library is not installed on your system, you should disable the
+SMT tests using the ``--no-smt`` option when running ``./scripts/tests.sh`` or
```

Comment:
I tested this but failed to disable SMT tests. I did

```
$ ./scripts/tests.sh --no-smt
...
test_0571770b984517916a9a795a2b5f8a37ce1c20f5eee047a30f3e5315718c32d4_s
ecurity_considerations_rst.sol
Unexpected compilation result:       
Expected failure: 0 - Expected warning / error output: 0
Was failure: 0

Warning: BMC analysis was not possible since no integrated SMT solver (
Z3 or CVC4) was found.
While calling:
...
```

Are you sure this flag works as intended?

Reply:
Adding the possibility is in this branch, did you try when checked out / built from this branch?

Reply:
Yes, I remember checking out the branch before I ran the command in my previous comment in this thread.

Reply:
Hmm, odd, it works for me. You generally have to rebuild everything to get tests correct, but I don't think they are affected by this… I can check on Linux I guess, but this just copied syntax I saw elsewhere in our bash scripts

Reply:
Okay, I'll check again (my system is Linux anyway).